Deploy-Policies.ps1

```diff
@@ -121,57 +121,145 @@ Import-Module -Name Microsoft.Graph.Beta.Groups
+if (-not $AADP2Group) { $AADP2Group = $Prefix + "_AADP2" }
+if (-not $SynchronizationServiceAccountsGroup) { $SynchronizationServiceAccountsGroup = $Prefix + "_Exclusion_SynchronizationServiceAccounts" }
+if (-not $RestrictedManagementAdministrativUnit) { $RestrictedManagementAdministrativUnit = $Prefix + "_RestrictedManagementAdministrativUnit" }
+if (-not $EmergencyAccessAccountsGroup) { $EmergencyAccessAccountsGroup = $Prefix + "_Exclusion_EmergencyAccessAccounts" }
+if (-not $RingTargeted) { $RingTargeted = $False }
+if (-not $RingGroup) { $RingGroup = $Prefix + "_" + $Ring }
+if (-not $AdministratorGroup) { $AdministratorGroup = $Prefix + "_Administrator" }
 #endregion
 
 #region functions
-function New-AFAzureADGroup($Name){
+
+#region function to generate or receive Entra ID Group
+function New-AFAzureADGroup {
+    [CmdletBinding()]
+    param (
+        $Name,
+        $RoleAssignable = $false
+    )
     $Group = Get-MgBetaGroup -Filter "DisplayName eq '$Name'"
-    if(-not $Group){
+    if (-not $Group) {
         Write-Host "Creating group: $Name"
-        $Group = New-MgBetaGroup -DisplayName $Name -SecurityEnabled:$true -MailEnabled:$false -MailNickname "NotSet" -Visibility Private -IsAssignableToRole:$true
+        $Group = New-MgBetaGroup -DisplayName $Name -SecurityEnabled:$True -MailEnabled:$false -MailNickname "NotSet" -Visibility Private -IsAssignableToRole:$RoleAssignable
     }
     Write-Host "ObjectId for $Name $($Group.Id)" 
     return $Group.Id
 }
+
+#region function for generate or receive Restricted Management Administrativ Unit
+function New-AFRMAU($Name) {
+    # Generate or receive Restricted Management Administrativ Unit
+    $RMAU = Get-MgBetaAdministrativeUnit -Filter "DisplayName eq '$Name'"
+    if (-not $RMAU) {
+        Write-Host "Creating RMAU: $Name"
+        $RMAU = New-MgBetaAdministrativeUnit -DisplayName $Name -Description "RMAU for Conditional Access Policies" -IsMemberManagementRestricted:$True
+    }
+    Write-Host "ObjectId for $Name $($RMAU.Id)" 
+    return $RMAU.Id
+}
+#endregion 
+
+#region Add Member to RMAU
+function Add-AFMembertoRMAU {
+    [CmdletBinding()]
+    param (
+        $RMAUID,
+        $GroupID
+    )
+    Write-Host "Check if group $GroupID is already member of RMAU $RMAUID"
+    $RMAUMemberAssign = Get-MgBetaAdministrativeUnitMember -AdministrativeUnitId $RMAUID -Filter "id eq '$GroupID'" -ErrorAction SilentlyContinue
+
+    if (-not $RMAUMemberAssign) {
+        Write-Host "Adding group $GroupID to RMAU $RMAUID"
+        $params = @{
+            "@odata.id" = "https://graph.microsoft.com/v1.0/groups/{$GroupID}" 
+        }
+        New-MGBetaAdministrativeUnitMemberbyRef -AdministrativeUnitId $OBjectID_RMAU -BodyParameter $params
+    }
+}
+#endregion
+
+#region function for Add New Admin Assignment for RMAU
+function Add-AFAdminAssignmenttoRMAU {
+    [CmdletBinding()]
+    param (
+        $RMAUID,
+        $RoleAssignmentID = 'db004403-d48a-49c3-a098-088c45470d56',
```

Comment:
What is this being hardcoded to, is it an off the shelve role? If so which one? @Timsto?

Reply:
in multiplte tenant is was an identityfier for the Group Administrator.  so there is a Group Admin Scope on the RMAU that the CA Admin can manage the group

Reply:
Thats good, That makes sense, cos I had to manually add myself as a group admin, thought I missed a step in the documentation. I'll do a PR.